src/mkdocstrings/inventory.py

```diff
@@ -0,0 +1,22 @@
+import zlib
+
+
+HEADER = """# Sphinx inventory version 2
+# Project: {project}
+# Version: {version}
+# The remainder of this file is compressed using zlib.
+"""
+
+
+def get_inventory(autorefs, project="project", version="0.0.0", compress=False):
+    header = HEADER.format(project=project, version=version).encode("utf8")
+
+    lines = []
+    for anchor, url in autorefs._url_map.items():
+        url = url.replace(anchor, "$")
+        lines.append(f"{anchor} py:obj 1 {url} -".encode("utf8"))
+
+    data = b"\n".join(lines)
+    if compress:
+        data = zlib.compress(data, 9)
+    return header + data
```

Comment:
So I've just implemented this version that doesn't use sphobjinv, and it seems to work fine.

Reply:
&lt;nod&gt;, looks pretty good to me.

If you don't mind adding Sphinx to your dev/test dependencies, there's a not-horribly-painful way to sanity-check whether Sphinx can read a generated inventory; see e.g. [here](https://github.com/bskinn/sphobjinv/blob/260a96cd4eaa7d5cdd8e12f03ae7376243841fa6/conftest.py#L163-L177).

Reply:
Sure, will do, too bad Sphinx itself is so heavy :sweat_smile: 

```
  • Installing pytz (2021.1)
  • Installing alabaster (0.7.12)
  • Installing babel (2.9.0)
  • Installing docutils (0.16)
  • Installing imagesize (1.2.0)
  • Installing requests (2.15.1)
  • Installing sphinxcontrib-applehelp (1.0.2)
  • Installing sphinxcontrib-devhelp (1.0.2)
  • Installing sphinxcontrib-htmlhelp (1.0.3)
  • Installing sphinxcontrib-jsmath (1.0.1)
  • Installing sphinxcontrib-qthelp (1.0.3)
  • Installing sphinxcontrib-serializinghtml (1.1.4)
  • Installing sphinx (3.5.2)
```

If you have any idea on how I could avoid all these contribs in the dev-deps, I'm all ears :slightly_smiling_face:

Reply:
@pawamoy In a quick test, I was able to use `sphinx.util.inventory.InventoryFile` successfully with just a `pip install sphinx docutils --no-deps`:

```
>>> from sphinx.util.inventory import InventoryFile
>>> with open("/path/to/objects_attrs.inv", "rb") as f:
...   InventoryFile.load(f, "", os.path.join)
...
{'py:class': {'attr.Attribute':  ... }}
```

Reply:
Nice! I'll see if I can set this pip install command as a task or something, as I won't be able to prevent Poetry from installing subdeps.

Reply:
Yep, should be possible to have that extra `pip install` just be an additional `run` step in your workflow. I do it all the time.